api/base/serializers.py

```diff
@@ -225,7 +225,7 @@ def to_internal_value(self, data):
             if request.method in utils.UPDATE_METHODS and not utils.is_bulk_request(request):
                 id_field = getattr(self.root.instance, self.source, '_id')
                 if id_field != data:
-                    raise Conflict()
+                    raise Conflict(detail=('The id you specified in the URL "' + id_field + '" exists, but does not match the id you specified in the JSON body "' + data + '".'))
```

Comment:
Also, are you just using your email notifications? Best to check the discussion; I had made some edits a minute or two after setting the original message that I think got lost. https://github.com/CenterForOpenScience/osf.io/pull/5776#discussion_r65555351